### PR TITLE
Chore versions upgrade

### PR DESCRIPTION
**Pull request by SimonDmz**
upgrade to spring-boot 3
dependencies upgrade
CVE fixes

### CODE REVIEW

File: src/main/java/fr/insee/pearljam/api/configuration/auth/KeycloakSecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+		jwtAuthenticationConverter
+				.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthoritiesConverter(roleProperties));
+		return jwtAuthenticationConverter;
+	}
+
+	Converter<Jwt, Collection<GrantedAuthority>> jwtGrantedAuthoritiesConverter(RoleProperties roleProperties) {
+		return new GrantedAuthorityConverter(roleProperties);
+	}
+
+	/**
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/configuration/auth/KeycloakSecurityConfiguration.java:103]: what to do with these ?

File: src/main/java/fr/insee/pearljam/api/domain/Person.java [hunk 2]
```diff
@@ -5,84 +5,80 @@
...
-* @author Corcaud Samuel
-* 
-*/
+ * Entity Person : represent the entity table in DB
+ * 
+ * @author Corcaud Samuel
+ * 
+ */
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/Person.java:27]: Lombok ?

File: src/main/java/fr/insee/pearljam/api/domain/OrganizationUnit.java [hunk 3]
```diff
@@ -3,22 +3,22 @@
...
-* @author Claudel Benjamin
-* 
-*/
+ * Entity OrganizationUnit : represent the entity table in DB
+ * 
+ * @author Claudel Benjamin
+ * 
+ */
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/OrganizationUnit.java:23]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/Comment.java [hunk 4]
```diff
@@ -2,23 +2,24 @@
...
-* 
-*/
+ * Entity Comment : represent the entity table in DB
+ * 
+ * @author Claudel Benjamin
+ * 
+ */
 @Entity
 @Table
 public class Comment implements Serializable {
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/Comment.java:25]: Lombok ?

File: src/main/java/fr/insee/pearljam/api/domain/ContactAttempt.java [hunk 5]
```diff
@@ -2,24 +2,24 @@
...
-* @author Claudel Benjamin, SimonDmz
-* 
-*/
+ * Entity ContactAttempt : represent the entity table in DB
+ * 
+ * @author Claudel Benjamin, SimonDmz
+ * 
+ */
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/ContactAttempt.java:24]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/ContactOutcome.java [hunk 6]
```diff
@@ -2,24 +2,24 @@
...
-* @author Claudel Benjamin
-* 
-*/
+ * Entity ContactOutcome : represent the entity table in DB
+ * 
+ * @author Claudel Benjamin
+ * 
+ */
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/ContactOutcome.java:24]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/Interviewer.java [hunk 7]
```diff
@@ -5,15 +5,15 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
 
 import fr.insee.pearljam.api.dto.interviewer.InterviewerContextDto;
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/Interviewer.java:19]: Lombok ?

File: src/main/java/fr/insee/pearljam/api/domain/Message.java [hunk 8]
```diff
@@ -3,75 +3,73 @@
...
-* 
-*/
+ * Entity Message : represent the entity table in DB
+ * 
+ * @author Paul Guillemet
+ * 
+ */
 
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/Message.java:28]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/MessageStatus.java [hunk 9]
```diff
@@ -2,37 +2,34 @@
...
-*/
+ * Entity MessageRecipient : represent the entity table in DB
+ * 
+ * @author Paul Guillemet
+ * 
+ */
 
 @Entity
 @Table
 public class MessageStatus implements Serializable {
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/MessageStatus.java:26]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/MessageStatusId.java [hunk 10]
```diff
@@ -3,43 +3,44 @@
...
 /**
  * This class is used to defines the association between OrganizationUnit and
  * Campaign tables.
+ * 
  * @author Guillemet Paul
  */
 @Embeddable
-public class MessageStatusId implements Serializable{
-	
+public class MessageStatusId implements Serializable {
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/MessageStatusId.java:16]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/OUMessageRecipient.java [hunk 11]
```diff
@@ -2,67 +2,63 @@
...
-*/
+ * Entity MessageRecipient : represent the entity table in DB
+ * 
+ * @author Paul Guillemet
+ * 
+ */
 
 @Entity
 @Table
 public class OUMessageRecipient implements Serializable {
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/OUMessageRecipient.java:23]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/OUMessageRecipientId.java [hunk 12]
```diff
@@ -3,42 +3,42 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
 
 /**
  * This class is used to defines the association between OrganizationUnit and
  * Campaign tables.
+ * 
  * @author Guillemet Paul
  */
 @Embeddable
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/OUMessageRecipientId.java:15]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/State.java [hunk 13]
```diff
@@ -2,58 +2,59 @@
...
-* @author Claudel Benjamin
-* 
-*/
+ * Entity State : represent the entity table in DB
+ * 
+ * @author Claudel Benjamin
+ * 
+ */
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/State.java:24]: lombok

File: src/main/java/fr/insee/pearljam/api/domain/User.java [hunk 14]
```diff
@@ -3,24 +3,24 @@
...
-* 
-*/
+ * Entity User : represent the entity table in DB
+ * 
+ * @author Corcaud Samuel
+ * 
+ */
 @Entity
-@Table(name="user", schema="public")
+@Table(name = "user", schema = "public")
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/User.java:23]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/Visibility.java [hunk 15]
```diff
@@ -2,75 +2,103 @@
...
-* @author Corcaud Samuel
-* 
-*/
+ * Entity Visibility : represent the entity table in DB
+ * 
+ * @author Corcaud Samuel
+ * 
+ */
 @Entity
 @Table
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/Visibility.java:21]: Lombok

File: src/main/java/fr/insee/pearljam/api/domain/VisibilityId.java [hunk 16]
```diff
@@ -2,44 +2,46 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
 
 /**
  * This class is used to defines the association between OrganizationUnit and
  * Campaign tables.
+ * 
  * @author scorcaud
  */
 @Embeddable
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/domain/VisibilityId.java:14]: Lombok

File: src/main/java/fr/insee/pearljam/api/dto/campaign/CampaignDto.java [hunk 17]
```diff
@@ -74,10 +74,10 @@ public CampaignDto(String id, String label, String email, IdentificationConfigur
 
 	public void setCampaignStats(List<Object[]> obj) {
 		if (obj != null && !obj.isEmpty() && obj.get(0).length > 3 && obj.get(0)[0] != null) {
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/dto/campaign/CampaignDto.java:76]: strange

File: src/main/java/fr/insee/pearljam/api/dto/closingcause/ClosingCauseCountDto.java [hunk 18]
```diff
@@ -27,14 +26,14 @@ public ClosingCauseCountDto() {
 		super();
 	}
 
-	public ClosingCauseCountDto(Map<String, BigInteger> obj) {
+	public ClosingCauseCountDto(Map<String, Long> obj) {
 		boolean nullOrEmpty = Optional.ofNullable(obj.isEmpty()).orElse(true);
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/dto/closingcause/ClosingCauseCountDto.java:30]: strange behavior

File: src/main/java/fr/insee/pearljam/api/service/impl/SurveyUnitServiceImpl.java [hunk 19]
```diff
@@ -197,14 +198,15 @@ private void updateCommunicationRequest(SurveyUnit surveyUnit, SurveyUnitDetailD
 					.map(dto -> new CommunicationRequest(dto, surveyUnit)).collect(Collectors.toSet());
 			communicationRequests.addAll(newCommunicationsRequests);
 
+			// NOTE
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/service/impl/SurveyUnitServiceImpl.java:201]: what to do with this ?

File: src/main/java/fr/insee/pearljam/api/service/impl/UserServiceImpl.java [hunk 20]
```diff
@@ -180,7 +174,13 @@ public boolean checkValidity(UserDto user) {
...
 	@Override
 	public UserDto createUser(UserDto userToCreate) throws NotFoundException {
-		OrganizationUnit ou = organizationUnitRepository.getOne(userToCreate.getOrganizationUnit().getId());
+		Optional<OrganizationUnit> ouOpt = organizationUnitRepository
+				.findById(userToCreate.getOrganizationUnit().getId());
+		if (ouOpt.isEmpty()) {
+			throw new NotFoundException(String.format("Organization Unit with id %s not found",
+					userToCreate.getOrganizationUnit().getId()));
+		}
+		OrganizationUnit ou = ouOpt.get();
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/service/impl/UserServiceImpl.java:183]: OrganizationUnit ou = ouOpt.orElseThrow(new NotFoundException(...))

File: src/main/java/fr/insee/pearljam/api/web/exception/ExceptionControllerAdvice.java [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+
+    @ExceptionHandler(AuthenticationTokenException.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ResponseBody
+    public void authenticationTokenExceptionException(AuthenticationTokenException e, WebRequest request) {
+        log.error(e.getMessage(), e);
+        processException(e, HttpStatus.INTERNAL_SERVER_ERROR, request, ERROR_OCCURRED_LABEL);
+    }
+
+    // @ExceptionHandler(HabilitationException.class)
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/web/exception/ExceptionControllerAdvice.java:148]: to delete ?

File: src/main/java/fr/insee/pearljam/api/web/validation/IdValid.java [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+import static java.lang.annotation.ElementType.*;
+
+/**
+ * Annotation used to validate identifiers
+ */
+@Target({ FIELD, PARAMETER, METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Constraint(validatedBy = IdValidator.class)
+public @interface IdValid {
```
Comment (davdarras) [inline src/main/java/fr/insee/pearljam/api/web/validation/IdValid.java:20]: used in app ?

File: src/main/resources/application-dev.properties [hunk 23]
```diff
@@ -1,20 +1,21 @@
-fr.insee.pearljam.env=dev
+spring.config.import=optional:file:${HOME}/pearl-api/local.properties
```
Comment (davdarras) [inline src/main/resources/application-dev.properties:1]: delete this file

File: src/test/resources/db/changelog/000_init.xml [hunk 24]
```diff
@@ -1,249 +0,0 @@
-<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (davdarras) [inline src/test/resources/db/changelog/000_init.xml:1]: liquibase not used anymore ?

**Review by davdarras — CHANGES_REQUESTED** [review]
Some points to discuss: 
- refactor security urls (commented at this moment)
- no more liquibase ?

File: src/main/java/fr/insee/pearljam/api/web/validation/IdValid.java (hunk 22 above)
Comment (SimonDmz) [reply src/main/java/fr/insee/pearljam/api/web/validation/IdValid.java:20]: not yet 🙈

File: src/main/java/fr/insee/pearljam/api/dto/campaign/CampaignDto.java (hunk 17 above)
Comment (SimonDmz) [reply src/main/java/fr/insee/pearljam/api/dto/campaign/CampaignDto.java:76]: too much output cases and too few dto
better archi is the answer, to be continued

File: src/main/java/fr/insee/pearljam/api/dto/closingcause/ClosingCauseCountDto.java (hunk 18 above)
Comment (SimonDmz) [reply src/main/java/fr/insee/pearljam/api/dto/closingcause/ClosingCauseCountDto.java:30]: no concrete model for counts

**Review by davdarras — APPROVED** [review]

**Pull request merged**